Ahmedq02/task-app
# Migrate REST API

# Migrate REST API



## Description



- **Related Issue:** Closes #7 
- **Summary of Changes:**
  - Created REST APIs for CRUD operations
  - Migrated main app to consume the APIs
  - Migrated server to asgi

## How to Test



1. Check out this branch: `git checkout <branch-name>`
2. Apply migrations (if any): `make migrate`
3. Collect static filed: `poetry run python manage.py collectstatic`
4. Run the development server: `make run`
5. Test the following:
   - [x] Task creation works as expected
   - [x] Task status updates are correctly monitored
   - [x] UI displays task progress properly
   - [x] API endpoints (if applicable) respond with the correct data

## Deployment Notes



- [x] This PR requires database migrations
- [ ] New environment variables need to be added (describe if any)
